### ISSUE TITLE
[META] Allow non-GitHub users to "follow" project nodes and be notified of changes.

### ISSUE THREAD

**Issue by klonos** [backdrop-ops/backdropcms.org#59]
The watch/unwatch feature of GitHub repos is great for all us GitHub "veterans", but it has a few drawbacks that do not make it ideal for Backdrop CMS users...

For one, even the whole UI is developer-oriented with almost everything (terminology, layout, color theme and icons) not as user-friendly as backdropcms.org. Another thing is that regular Backdrop users would definitely appreciate getting all the information they need from a single point (backdropcms.org) and not having to subscribe to 3rd party websites.

Then, say a user already has a GitHub account. It's the fact that one can watch a whole repo, but if they do, they'll be "bombed" with email notifications about literally **everything** (new releases, new issues etc). One simply cannot "follow" only new releases, or just the README.md for example because such fine-grained notifications are not possible.

In d.org, we have this issue (still not implemented): [Add the ability to follow projects like we follow issues](https://www.drupal.org/node/1434450) that aims to put in place the same mechanism that is used in order to follow/unfollow issues (uses the Flag module if I'm not mistaken). I would like to propose to implement something like that in backdropcms.org that would notify the subscriber of:

- **New releases of a project**. What I currently do to work around this in Drupal is to keep projects in the modules or themes directories, but have them disabled. I then activate the "Check for updates of disabled modules and themes" setting (under ```/admin/reports/updates/settings```) and regularly (manually) check the *Available updates* page for any updates. Not automated and not ideal at all. D.org seems to offer the "View all releases" page for each project and they do have a "Subscribe with RSS" thing in that page, but that is a plain text - not a link (something missed from the d.org upgrade to D7 I guess) and it is located at the very end of the releases list below the pager:

![d org_project_releases_rss_link](https://cloud.githubusercontent.com/assets/2423362/7213242/c546241c-e5ba-11e4-8b6c-5a7653d952a5.png)

Not discoverable and not useful at all. We should allow people to subscribe and be notified of new releases for the projects the care about by email notifications/alerts. Use case is a person that is for example interested in a project, but cares only to know when a (stable/initial) release is out without having to install any unstable version (just to get notifications through the *Available updates* page) and without having to create a GitHub account and watch a whole repository.

We should also account for new projects without any releases yet (our version of dev modules). A person might want to follow a project repo because they have read what it aims to do, but they do not care about the issue queue or any other changes other than an initial release. I realize that project nodes in backdropcms.org are currently created on initial release (right?), so we need to make them be created on project addition to the contrib repo instead (with a "no release yet" note and a "subscribe to be notified when there is a release" button).

- **Updates in the project description** (with revision diff). As it is now, d.org does offer a "Revisions" tab for project nodes, but one needs to remember when they last visited the project page in order to compare and figure out what has changed since then. And of course there are no notifications through email, so you have to "magically" guess that a change has occurred in the first place. No good. We should offer the same revisions functionality with the added bonus of remembering the last revision the user had viewed last time and compare that to the current revision by default. Email notifications of project description changes too (linking to the diff page).

- **Gist of known issues**. Just an email with a list of the currently open issues (with links to GitHub). Separate sections with new issues filed since last email notification and with old ones still open as well as the ones closed since last email (version resolved in added as side-info).

All this subscription feature should be granular allowing people to select which of the things they care about they should be notified for (releases, project mete-data, project issues).

A lot of work, I understand. So this is a meta-issue. No child issues created yet, lets discus first.

**Comment by quicksketch** [comment]
Definitely a lot of work described here. :smile: It's good to have direction and goals so I'm glad we're thinking about this. Note that right now we don't have any kind of notifications system set up at all, so this probably has some deeper implications. I filed a new issue to track user logins, which we'll obviously need before users can do any of this: https://github.com/backdrop-ops/backdropcms.org/issues/61.

**Comment by klonos** [comment]
Yep, thinking ahead here. So first things first, lets get the user login done.

**Comment by BWPanda** [comment]
> It's the fact that one can watch a whole repo, but if they do, they'll be "bombed" with email notifications about literally everything (new releases, new issues etc). One simply cannot "follow" only new releases, or just the README.md for example because such fine-grained notifications are not possible.

Note that GitHub has since implemented the ability to watch a repo and be notified of only new releases: 
![image](https://user-images.githubusercontent.com/2385329/77198387-8d629e80-6aef-11ea-852f-091c714d048f.png)
